pkgs/applications/editors/vim/plugins/overrides.nix

```diff
@@ -960,6 +960,34 @@ self: super: {
     ];
   };
 
+  vim-graphical-preview = super.vim-graphical-preview.overrideAttrs (old:
+    let
+      vim-graphical-preview-rust = rustPlatform.buildRustPackage {
+        pname = "vim-graphical-preview-rust";
+        inherit (old) version src;
+
+        cargoHash = "sha256-cMMNur6QKp87Q28JyCH2IMLE3xDVd7Irg9HvJ2AsnZc=";
+
+        nativeBuildInputs = [ pkg-config ];
+
+        buildInputs = [ openssl ] ++ lib.optionals stdenv.isDarwin [
+          darwin.apple_sdk.frameworks.Security
+        ];
+
+        cargoBuildFlags = [ "--workspace" ];
+
+        # tests are broken
+        doCheck = false;
+      };
+    in
+    {
+      dependencies = with self; [ plenary-nvim ];
+      postInstall = ''
+        mkdir -p $out/target/debug
+        ln -s ${vim-graphical-preview-rust}/{bin,lib}/* $out/target/debug
+      '';
+    });
```

Comment:
please don't just copy from sg-nvim and actually try to build it, vim-graphical-preview seems to have a different set of dependencies

Reply:
Sorry, I'm still learning Nix things. I'm going to remove that plugin for now till I get some more time to figure things out better. 

I think I squashed everything down correctly this time maybe.. 

Thank you